keyboards/moonlander/matrix.c

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2020 ZSA Technology Labs, Inc <@zsa>
+Copyright 2020 Jack Humbert <jack.humb@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include "hal.h"
+#include "timer.h"
+#include "wait.h"
+#include "printf.h"
+#include "matrix.h"
+#include "action.h"
+#include "keycode.h"
+#include <string.h>
+#include "moonlander.h"
+#include "i2c_master.h"
+#include "debounce.h"
+
+/*
+#define MATRIX_ROW_PINS { B10, B11, B12, B13, B14, B15 } outputs
+#define MATRIX_COL_PINS { A0, A1, A2, A3, A6, A7, B0 }   inputs
+ */
+/* matrix state(1:on, 0:off) */
+extern matrix_row_t matrix[MATRIX_ROWS];
+static matrix_row_t matrix_debouncing[MATRIX_ROWS];
+static matrix_row_t matrix_debouncing_right[MATRIX_COLS];
+static bool         debouncing            = false;
+static uint16_t     debouncing_time       = 0;
+static bool         debouncing_right      = false;
+static uint16_t     debouncing_time_right = 0;
+
```

Comment:
```suggestion
```

Reply:
Not pulling out the debounce code.   It's debouncing the left and right halves separately, and would prefer to keep it that way, because the MCP chip is ... not exactly fast.

However, I have set the debouncing type to "custom", so it shouldn't be an issue.